README.md

```diff
@@ -1,39 +1,167 @@
-
-Exercise 2: finishing up the old work
-1. After your changes are successfully reviewed, revised, and pulled in, it is
-time to clean up your stuff.
-1. Synchronize the master branch of your local clone with upstream master
-1. Delete the branch you used for the finished.txt update
-1. In a good old-fashioned email, write a brief account of everything you did to
-complete Exercise 1 and 2 and why you did it (i.e., what purpose did each step
-serve)
+# Git workflow
+
+This page is aimed to be a simplified version of github help. For complete workflow,
+please go to [full git documentation](https://guides.github.com/). Also Google and [stackoverflow](https://stackoverflow.com/) is incredibly helpful.
+
+Below is a code block for reference. It includes the most basic git commands you will
+use to complete the workflow introduced below. If it's your first time
+here, skip this block and finish reading the workflow doc first.
+```git
+git clone <remote url>  # clone a repo
+git remove add <name> <remote url>  # add a remote
+git remove -v  # list all remotes
+git remote update  # update remotes
+git branch -a  # list all available branches
+git checkout <branch name>  # checkout to available branch
+git checkout -b <new branch name> # checkout to new branch based on current branch
+git add <path to the file>  # add changes into working history
+git commit  # commit changes
+git push <remote name>  <branch name>  # push local to remote
+```
+# Step 0: get all required tools and reference
+1. First you need a [github account](https://github.com/).
+1. You would also need to have [git](https://git-scm.com/) installed on your computer.
+1. Read chapters 15 and 16 of **Effective Computation in Physics** by
+Scopatz and Huff (it is strongly recommended that you get a copy, but we do have).
+1. We also recommend [PyCharm](http://www.jetbrains.com/pycharm/) because
+  * it's powerful.
+  * it's widely used in the group, so you can get help easily.
+  * as students you can request a free copy of the full production version!
+
+# Step 1: Fork the repository you wish to contribute
+Forking means you keep the entire copy of certain github repository under your account. 
+The advantages of forking are:
+1. it doesn't affect the original repository, i.e. what was working remains working.
+1. now this repository is completely under your account, you can do whatever you want. `fork` can be done through github website:
+
+![alt text](https://github.com/chiahaoliu/group_git_workflow/blob/doc_gitworkflow/img/git_fork.png "")
+
+# Step 2: Clone the git repository to local machine
+1. Go to the repository you wish to work on (ideally under your fork). Click on the green **clone or download** button at the top-right corner:
+
+![alt text](https://github.com/chiahaoliu/group_git_workflow/blob/doc_gitworkflow/img/git_clone.png "")
+
+1. The link shows here is the identity to this repository. You can then copy it and paste to your terminal and do:
```

Comment:
you don't copy and paste, you 
" * click on the "clipboard icon" to copy the link to your clipboard."